activestorage/lib/active_storage/engine.rb

```diff
@@ -24,7 +26,7 @@ class Engine < Rails::Engine # :nodoc:
 
     config.active_storage = ActiveSupport::OrderedOptions.new
     config.active_storage.previewers = [ ActiveStorage::Previewer::PopplerPDFPreviewer, ActiveStorage::Previewer::MuPDFPreviewer, ActiveStorage::Previewer::VideoPreviewer ]
-    config.active_storage.analyzers = [ ActiveStorage::Analyzer::ImageAnalyzer, ActiveStorage::Analyzer::VideoAnalyzer ]
+    config.active_storage.analyzers = [ ActiveStorage::Analyzer::ImageAnalyzer::Vips, ActiveStorage::Analyzer::ImageAnalyzer::ImageMagick, ActiveStorage::Analyzer::VideoAnalyzer ]
```

Comment:
This looks like a breaking change. What happens on this branch if you try to use `ActiveStorage::Analyzer::ImageAnalyzer` directly?

Reply:
Yes, it is. I didn't think about that. I can return my previous code, but it was looking pretty ugly:

```
def metadata
  if ActiveStorage.variant_processor == :vips
    Analyzer::ImageAnalyzer::Vips.new(@blob).metadata
  else
    Analyzer::ImageAnalyzer::ImageMagick.new(@blob).metadata
  end     
end
```

It also does not fit the pattern that was set with the PDF previewers, where each lib is a different previewer in the array, instead of having a generic `PDFPreviewer` that then chooses the correct class based on the libs available:
```
config.active_storage.previewers = [ 
  ActiveStorage::Previewer::PopplerPDFPreviewer, 
  ActiveStorage::Previewer::MuPDFPreviewer, 
  ActiveStorage::Previewer::VideoPreviewer 
]
```

I'm not sure where to go from here: breaking change, different patterns, or changing the previewers to have a generic `PDFPreviewer` that then decides between `mu` and `poppler`.

Reply:
I've also just noticed I am not respecting the naming pattern and should probably have named the analyzers `VipsImageAnalyzer` and `ImageMagickImageAnalyzer` so they match the way the previewers are named.

Reply:
Or I could return the original `ImageAnalyzer` and just add the VipsVersion ahead of it:
```
config.active_storage.analyzers = [ 
  ActiveStorage::Analyzer::VipsImageAnalyzer, 
  ActiveStorage::Analyzer::ImageAnalyzer, 
  ActiveStorage::Analyzer::VideoAnalyzer 
]
```

This would not be a breaking change and it would more or less follow the way the previewers are set. I'd just need another class to place the common `metadata` method so it stayed DRY.

Reply:
I think the approach here is good, and if users have been doing things like `config.active_storage.analyzers << MyAnalyzer` then it shouldn't cause problems. The risk I can think of is code like `config.active_storage.analyzers.insert(config.active_storage.analyzers.index(ActiveStorage::Analyzer::ImageAnalyzer) + 1, MyAnalyzer)` breaking. Which seems like a bad pattern anyway.

Can you please update the docs: https://github.com/rails/rails/blob/main/guides/source/configuring.md#configuring-active-storage

Reply:
Done. I've also updated the comments in the code. Unfortunately, the CI failed on the same place as last time (the bug report templates I think)

Reply:
you might need @zzak to approve it again

Reply:
@brenogazzola @ghiculescu Approved! :pray: